workspace/run_browser_tests.sh

```
@@ -103,7 +103,14 @@ echo "✅ Server ready"
 # 6. Run Playwright browser tests using native headless mode
 echo "🧪 Running Playwright Browser Tests target: $TEST_TARGET ($SUITE_NAME)..."
 mkdir -p "$PROJECT_DIR/artifacts"
-source /opt/venv/bin/activate
+if [ -f "/opt/venv/bin/activate" ]; then
+  source /opt/venv/bin/activate
```

Shellcheck (0.11.0)

[info] Not following: /opt/venv/bin/activate was not specified as input (see shellcheck -x).

(SC1091)